lib/sensu/transport/redis.rb

```diff
@@ -0,0 +1,307 @@
+          info = {
+            :messages => messages,
+            :consumers => 0
+          }
+          callback.call(info)
+        end
+      end
+
+      private
+
+      # Reset instance variables, called when reconnecting.
+      def reset
+        @connections = {}
+      end
+
+      # Return or setup a named Redis connection. This method creates
+      # a Redis connection object using the provided Redis transport
+      # options. Redis auto-reconnect is disabled as the connection
+      # "pool" is monitored as a whole. The transport `@on_error`
+      # callback is called when Redis errors are encountered.
+      #
+      # @param name [String] the Redis connection name.
+      # @return [Object]
+      def redis_connection(name)
+        return @connections[name] if @connections[name]
+        connection = EM::Protocols::Redis.connect(@options)
+        connection.auto_reconnect = false
+        connection.reconnect_on_error = false
+        connection.on_error do |error|
+          @on_error.call(error)
+        end
+        @connections[name] = connection
+        connection
+      end
+
+      # Monitor current Redis connections, the connection "pool". A
+      # timer is used to check on the connections, every `3` seconds.
+      # If one or more connections is not connected, a forced
+      # `reconnect()` is triggered. If all connections are connected
+      # after reconnecting, the transport `@after_reconnect`
+      # callbacked is called. If a connection monitor (timer) already
+      # exists, it is canceled.
+      def monitor_connections
+        @connection_monitor.cancel if @connection_monitor
+        @connection_monitor = EM::PeriodicTimer.new(3) do
+          if !connected?
+            reconnect(true)
+          elsif @reconnecting
+            @after_reconnect.call
+            @reconnecting = false
+          end
+        end
+      end
+
+      # Create a Redis key within the defined Redis keyspace. This
+      # method is used to create keys that are unlikely to collide.
+      #
+      # @param type [String]
+      # @param name [String]
+      # @return [String]
+      def redis_key(type, name)
+        [REDIS_KEYSPACE, type, name].join(":")
+      end
+
+      # Publish a message to a Redis channel (PubSub). The
+      # `redis_key()` method is used to create a Redis channel key,
+      # using the transport pipe name. The publish callback info
+      # includes the current subscriber count for the Redis channel.
+      #
+      # http://redis.io/topics/pubsub
+      #
+      # @param pipe [String] the transport pipe name.
+      # @param message [String] the message to be published to the transport.
+      # @yield [info] passes publish info to an optional callback/block.
+      # @yieldparam info [Hash] contains publish information.
+      # @yieldparam subscribers [String] current subscriber count.
+      def pubsub_publish(pipe, message, &callback)
+        channel = redis_key("channel", pipe)
```

Comment:
pubsub ignores db. A publish to db 1 will be seen by a subscriber in db 10.

The lists can be managed by using different DBs, but that is then up to configuration management.